### DISCUSSION TITLE
Button Text Alignment

### DISCUSSION THREAD

**Discussion by AngeloKalpakis** [israel-dryer/ttkbootstrap#261] [Q&A] [answered]
Is it possible to align text within a button to the left or right within the button? Currently, it only defaults to the Center of the button. Button also contains an image. Buttons display as follows:

![image](https://user-images.githubusercontent.com/67386835/164070200-8425b12f-4528-4938-8072-ebb7303d002d.png)

But would like to have the button text to the Left.

P.S. Don't worry about the vertical line between the image and button text. I was experimenting with options.

**Comment by israel-dryer** [comment, marked as answer]
Yes. There are a few ways to do this. I'll show you the custom style option below, but I'll list a few other options.

- You can use a Label widget instead of a button and then add a \<Button-1\> binding to the label.
- You can use a Button widget and apply a label style `style="TLabel"`
- You can use the bootstyle "link" to make the button appear as a hyperlink... then make sure you do not allow the button to expand when you pack it to give the impression that it is left-aligned... ie. do not use `widget.pack(expand=YES, fill=X, side=LEFT)`.. instead, just use `widget.pack(side=LEFT)`.
- You can create a custom button style that anchors the text to the right or left.

Since the text anchoring is controlled in the style for a ttk button, you will need to create a custom button style. This means using the underlying ttk style that ttkbootstrap uses in the background.

For a button with a primary color, you would use the style `primary.TButton`. So, to subclass this style in ttk, you would just prefix that with your own unique name such as `my.primary.TButton`. 

```python
import ttkbootstrap as ttk
from ttkbootstrap.constants import *
from pathlib import Path


window = ttk.Window(themename="superhero", size=(300, 300))

# configure custom button style
window.style.configure('my.primary.TButton', anchor=W)

# create image
img_path = Path(__file__).parent / 'study.png'
img = ttk.PhotoImage(file=img_path)

# add widget container
frame = ttk.Frame(window, padding=15)
frame.pack(fill=BOTH, expand=YES)

# add widgets to container
ttk.Label(frame, image=img).pack(side=LEFT)
ttk.Button(frame, text="Library", style='my.primary.TButton').pack(side=LEFT, fill=X, expand=YES)

window.mainloop()
```

<img width="152" alt="image" src="https://user-images.githubusercontent.com/42584742/164127094-692d9fd2-924e-4117-8584-eae91d8342ee.png">

**Comment by AngeloKalpakis** [comment]
Outstanding. Thanks for this!
